NMDSdevopsServiceAdm/DataEngineering
# Move Job calculation logic to make adding new columns easier

https://trello.com/c/zDRXIoIE

# Description
To support data analysis we need to add new columns to the resultant data frame. This would be hard given the Job calculation logic is stored in one block. The code should be moved and made modular with test unit coverage to make that process easier and allow the task to be spread amongst the team. 

It will also allow us to write extra tests to understand the rule sets and make it easier to switch rule sets in an out as required in the future 
# How to test
- tests have moved
- new code has test coverage
- build is successful
- adapt script in pipeline to show row count for prepare locations and confirm it is the same as it is in the prod data set prepare locations# Developer checklist
